config/plugins/interactive_environments/cellxgene/templates/cellxgene.mako

```diff
@@ -0,0 +1,39 @@
+<%namespace name="ie" file="ie.mako" />
+<%
+    ie_request.load_deploy_config()
+
+    # Make the dataset available inside the container
+    data_file = ie_request.volume('/input/file.' + hda.ext, hda.file_name, mode='ro')
```

Comment:
I'm not sure this will work on usegalaxy.org. I think @natefoo disabled the mount option and prefers to download the files via the API into the container.

Reply:
Is that true @natefoo? Seems much less efficient.

Reply:
@natefoo doesn't always respond to Github pings 😅, my understanding is that this is the case and @bgruening is correct - we don't mount the usegalaxy.org object store on the Docker hosts.

Reply:
Oh no, is this open to discussion? ;)

I'm hoping we can develop a really simple contract for containers to serve as IEs. A lot of the boilerplate could go away. One nice thing about mounting data into the container is this container is completely Galaxy agnostic. If I need to use the API I have to install a bunch of Galaxy specific stuff in the container...

Reply:
It's a limitation of using Swarm (or really any Docker server that's not data-local).

Reply:
So should we not support volume mounts then?

Reply:
It's not a bad solution if your container is data-local since you don't have to copy, but it can't be the only access method unless we e.g. use @erasche's Galaxy Docker storage driver in production (which I think is a good idea if it can be made production quality). Currently the IEs we use support both mounts and copying on startup.

Reply:
Yeah, that was my plan, volume mounts are quite messy and requires supporting two very different behaviours and trying to make it appear seamless, so just throw all the current stuff away, add a docker volume driver, and whenever your IE starts, it has an auto-attached volume with all of that user's datasets (virtually), and you can just read from paths like normal.

It's sub-optimal in the data local to container situation, and we'll need to find a way to support that setup somehow as well. It probably will be "turn on expose dataset paths", and then the mounted volume will just use symlinks, and you'll be responsible that all data is mounted in somehow.

Reply:
Yeah, I really like your solution @erasche. For the moment, removing all the volume mounts and only using the API seems like the right step, it would simplify IE development. I'd also suggest we always set DATASET_ID and DATASET_IDS environment variables so individual IEs don't have to do so. Less boilerplate.

(I'll try to fix this up, unfortunately I've not yet been able to get `galaxy_ie_helpers` working under Docker Mac for local testing).

Reply:
Yes, exactly the idea, but unfortunately not implemented enough that we can ask you to make those changes now, we'll have to do them in another round. If you need this for something other than a usegalaxy.*, then we should probably merge as-is, and fix later?